### PR TITLE
🏃 add e2e framework to CI build

### LINKED ISSUE #1772: test/framework isn't being build by CI
**What steps did you take and what happened:**
test/framework currently doesn't compile (see #1771 for the fix) but CI (testgrid) is green.

**What did you expect to happen:**
CI failure

**Anything else you would like to add:**
test/framework is a separate go module, and it's not built by `scripts/ci-build.sh`

/kind bug
/help
/milestone v0.3.0
/priority important-soon

### PR DESCRIPTION
**Pull request by jadarsie**
Signed-off-by: Javier Darsie jadarsie@microsoft.com

**What this PR does / why we need it**:

~~I think the build is broken, this PR fixes it~~
~~Last commit fixes #1772~~

Fixes #1772

### CODE REVIEW

**Comment by ncdc** [comment]
Note, I checked out your branch and get this error with go1.12:

```
$ cd test/framework
$ go build -v ./...
sigs.k8s.io/cluster-api/test/framework/generators
# sigs.k8s.io/cluster-api/test/framework/generators
generators/capi.go:57:39: not enough arguments in call to kustomize.Run
        have ()
        want (context.Context)
note: module requires Go 1.13
sigs.k8s.io/cluster-api/test/framework
```

**Comment by jadarsie** [comment]
> Note, I checked out your branch and get this error with go1.12:
> 
> ```
> $ cd test/framework
> $ go build -v ./...
> sigs.k8s.io/cluster-api/test/framework/generators
> # sigs.k8s.io/cluster-api/test/framework/generators
> generators/capi.go:57:39: not enough arguments in call to kustomize.Run
>         have ()
>         want (context.Context)
> note: module requires Go 1.13
> sigs.k8s.io/cluster-api/test/framework
> ```

Yep, I missed a few more errors, will push the fix

**Comment by chuckha** [comment]
These are getting fixed here:

https://github.com/kubernetes-sigs/cluster-api/pull/1742

**Comment by chuckha** [comment]
@jadarsie can you squash this down to just the last commit? That can close out #1772

**Comment by jadarsie** [comment]
> @jadarsie can you squash this down to just the last commit? That can close out #1772

Done, will update the PR title

File: Makefile [hunk 1]
```diff
@@ -105,6 +106,10 @@ $(CONVERSION_GEN): $(TOOLS_DIR)/go.mod
 $(RELEASE_NOTES) : $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && go build -o $(RELEASE_NOTES_BIN) -tags tools ./release
 
+.PHONY: e2e-framework
+e2e-framework:
```
Comment (chuckha) [inline Makefile]: add a comment to this so it shows up in help

something like

```
e2e-framework: ## builds the capi e2e framework
```

**Comment by jadarsie** [comment]
> squashed

I see, sorry, I thought you were just proposing one alternative.

**Comment by chuckha** [comment]
/approve

This looks great, thank you

**Pull request merged**